docs/v1/tel-field.md

```diff
@@ -33,7 +33,7 @@ validation: (value: FieldValue, data: FormData, required: boolean) => true | str
 **Default value**
 
 ```js
-public telFormat = /^\+[1-9]\d{1,14}$/;
+public telFormat = /^\d{7,15}$/;
```

Comment:
This might just allow any random number. 
This means, string must consist solely of 7 to 15 digits, with no other characters allowed, and must match from the beginning to the end of the string for it to be considered valid.

Shall we really just allow any number with that length? 
This also removes/restricts users to not use country codes, which was asked in the ticket, agreed, but this might give them error or block them from adding any country code like `+56-3838....`

Reply:
Hi, it would be best that we allow optional plus at the start of the string, for the rest I think 7 to 15 digits is OK.

I created [this issue](https://github.com/form-js/forms.js/issues/73) that will allow more customization and would split the international codes to separate input so the cases regarding that will be handled in this issue. But still would allow the plus at the start just to make sure it is possible to input numbers with international code.

Thank you

Reply:
I agree with your point, in that case we could use `public telFormat = /^(\+?\d{1,3}-?)?\d{7,15}$/;`. it allows an optional +, 3 numbers and - symbol before the phone number.

Reply:
I feel like there is still a lot of international numbers that this could validate as invalid.

I have looked at how other resolved it and came by this post 

https://www.google.com/amp/s/www.geeksforgeeks.org/validate-phone-numbers-with-country-code-extension-using-regular-expression/amp/

Aditionally this Is what chatGPT reconended.

/^\+?(\d{1,3})?[-.\s]?\(?\d{1,4}?\)?[-.\s]?\d{1,4}[-.\s]?\d{1,4}[-.\s]?\d{1,9}$/

This pattern provides more flexibility:

^\+? matches an optional + at the beginning.
(\d{1,3})? matches an optional country code of 1 to 3 digits.
[-.\s]? allows an optional separator (hyphen, period, or space).
\(?\d{1,4}?\)? matches an optional area code that can be enclosed in parentheses.
[-.\s]? allows an optional separator.
\d{1,4} matches groups of digits (1 to 4 digits in this case) that can be repeated up to a total length of the phone number.
The overall length of the phone number can be between 7 and 15 digits, but grouped more flexibly.
Here are some examples that this regex would match:

+1 123-456-7890
+91-98765-43210
1234567890
(123) 456-7890
+44 20 7946 0958

We need a broad validation that checks basic phone formatting but we do not want to restrict anything. 

What do you think?

Reply:
That would work as well, but this could go on forever, theres always going to be new ways of formatting phone numbers that we can't account for, however the regex does eliminate the few main ones. We could use this.

I would suggest either adding a required format for users via a placeholder/error message, or have 2 inputs for the country code and phone number, they could be stored seperately as 2 different variables or could be joined together.

We could also look at using a library to validate phone numbers.

Reply:
I agree you are right for sure, I would prefer not to include any libraries lets make it as simple as possible, but still make sure it validates some kind of telephone number format.

- allow + on the start with 1-3 digits of country code a
- allow separator and after that a section that can be enclosed in parentheses
- optional separator again
- and then 1 to 9 digits possibly separated by commas or spaces

This should do the trick

`/^\+?\d{1,3}?[-.\s]?\(?\d{1,4}?\)?[-.\s]?\d{1,9}([-.\s]?\d{1,9})*$/`

i am still counting that we will add the possibility to have the country code as separate input.